### PR TITLE
iio: Support wildcard in IIO path input

### PR DESCRIPTION
**Pull request by ceolin**
It allows using wildcard characters to select an IIO path. It's
important to know that only the first result is used.

It internally uses glob().

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

### CODE REVIEW

**Comment by ceolin** [comment]
@fulong82 is it that you have in mind on https://github.com/solettaproject/soletta/issues/2154 ?

I've scratched it but I was not able to test yet. I'm creating this pullreq to see whether I understood correctly or not .

**Comment by fulong82** [comment]
@ceolin, 

Yes. So base on your pull request implementation, you are taking the 1st matching entry to proceed.
Thanks.

File: src/lib/io/sol-iio.c [hunk 1]
```diff
@@ -1572,6 +1573,33 @@ resolve_absolute_path_cb(void *data, const char *dir_path, struct dirent *ent)
...
+                errno = ENOMEM;
+                return NULL;
+            case GLOB_ABORTED:
+            default:
+                errno = EINVAL;
+                return NULL;
+        }
+    }
+
+    resolved_path = realpath(result.gl_pathv[0], resolved_path);
```
Comment (fulong82) [inline src/lib/io/sol-iio.c]: Isn't this should be:
`resolved_path = realpath(result.gl_pathv[0], *resolved_path);`

Comment (ceolin) [inline src/lib/io/sol-iio.c]: nope,  realpath() receives a pointer to an allocated memory and change this memory. But there is a problem in this code, sending a new pr. @fulong82 are you able to test it ?

**Comment by fulong82** [comment]
@ceolin 

Okay, tested your patch. It is working. Thx. 👍

**Comment by ceolin** [comment]
@fulong82 thanks for testing. Is it address the issue https://github.com/solettaproject/soletta/issues/2154 ?

**Comment by fulong82** [comment]
@ceolin 
Yes and Thanks.

**Pull request merged**